src/rpdk/core/jsonutils/utils.py

```diff
@@ -103,6 +103,97 @@ def traverse(document, path_parts):
     return document, tuple(path), parent
 
 
+# pylint: disable=R0913,C0301
+# flake8: noqa=C901,B950
+def traverse_list_safe(
+    document,
+    path_parts,
+    parent=None,
+    path=None,
+    paths=None,
+    documents=None,
+    parents=None,
```

Comment:
I don't understand what all these arguments are or why they are needed.  In general, less arguments is better.  like document/documents, parent/parents, path/paths/path_parts... can these all be combined? if you need say that parent = parents[0] or whatever in the method thats better than passing all as parameters.

Also, why are you setting the default value to None if you're then just checking for None and setting to empty array?

Reply:
The reason why these were added are because we are using a recursive approach. The singular variables like parent/path/document denote the inputs we are getting in this method while, the plural variables like documents/parents/paths denotes the list of outputs from this method. 
Pytest doe snot thing [] as default is a safe value to assign and expects None to be assigned. This also ensures the parameter list is empty every time this method is called.